server/athenian/api/controllers/filter_controller.py

```diff
@@ -105,13 +112,42 @@
     if status is None:
         raise ResponseError(ForbiddenError(
             detail="User %s is forbidden to access account %d" % (uid, filt.account)))
-    if not filt._in:
+    if not filt.in_:
         rss = await load_account_reposets(
             sdb_conn, mdb_conn, filt.account, native_uid, [RepositorySet.id])
-        filt._in = ["{%d}" % rss[0][RepositorySet.id.key]]
+        filt.in_ = ["{%d}" % rss[0][RepositorySet.id.key]]
     repos = set(chain.from_iterable(
         await asyncio.gather(*[resolve_reposet(r, ".in[%d]" % i, sdb_conn, uid, filt.account)
-                               for i, r in enumerate(filt._in)])))
+                               for i, r in enumerate(filt.in_)])))
     prefix = "github.com/"
     repos = [r[r.startswith(prefix) and len(prefix):] for r in repos]
     return repos
+
+
+async def filter_prs(request: AthenianWebRequest, body: dict) -> web.Response:
+    """List pull requests that satisfy the query."""
+    filt = FilterPullRequestsRequest.from_dict(body)
+    try:
+        repos = await _resolve_repos(
+            request.sdb, request.mdb, filt, request.uid, request.native_uid)
+    except ResponseError as e:
+        return e.response
+    stages = set(getattr(Stage, s.upper()) for s in filt.stages)
+    participants = {getattr(ParticipationKind, k.upper()): v
+                    for k, v in body.get("with", {}).items()}
```

Comment:
The built-in enum validation guarantees that there are only matching attrs.

Reply:
What do you mean? If the second value you pass to `getattr` is not a valid enum that would give you `None`, doesn't it?

Reply:
So I am saying that the framework's validation before executing this endpoint handler takes care of checking for invalid enum values, hence this is not possible. The mapping is 1-1, otherwise, it is a bug in the code that should pop up in Sentry asap.